libraries/AP_LQR_Control/AP_LQR_Control.cpp

```diff
@@ -259,16 +259,16 @@ void AP_LQR_Control::update_waypoint(const struct Location &prev_WP, const struc
     //Calculate minimum turning radius as function of groundspeed
     float min_turn_rad = 2*groundSpeed;
     
-    if (_q1_val != 0) {
+    if !is_zero(_q1_val) {
         q1 = _q1_val*0.01;
     }
     
-    if (_q1_lim > 0) {
+    if is_positive(_q1_lim) {
         q1 = MIN(q1, _q1_lim * 0.01);
     }
     
-    if (_q1_lim == -1) {
-        float q1_lim = sqrtf(float((_max_xtrack*0.01)/((_max_xtrack*0.01)-min_turn_rad)));
+    if is_negative(_q1_lim) {
+        float q1_lim = sqrtf(float((_max_xtrack*0.01)/(fabsf((_max_xtrack*0.01)-min_turn_rad))));
```

Comment:
I would say that taking absolute value of this difference is a bad idea, as if it is negative, the limitation looses any sense. Another concern is that _max_xtrack will mostly always be smaller than min_turn_rad.
And finally, the limitation did not solve the problem with flyaway, so most probably can be removed?